ledger/complete/wal/wal.go

```diff
@@ -49,20 +49,23 @@ func (w *DiskWAL) UnpauseRecord() {
 	w.paused = false
 }
 
-func (w *DiskWAL) RecordUpdate(update *ledger.TrieUpdate) error {
+func (w *DiskWAL) RecordUpdate(update *ledger.TrieUpdate) (segmentNum int, skipped bool, err error) {
 	if w.paused {
-		return nil
+		return 0, true, nil
 	}
 
 	bytes := EncodeUpdate(update)
 
-	_, err := w.wal.Log(bytes)
+	locations, err := w.wal.Log(bytes)
 
 	if err != nil {
-		return fmt.Errorf("error while recording update in LedgerWAL: %w", err)
+		return 0, false, fmt.Errorf("error while recording update in LedgerWAL: %w", err)
+	}
+	if len(locations) != 1 {
+		return 0, false, fmt.Errorf("error while recording update in LedgerWAL: got %d location, expect 1 location", len(locations))
```

Comment:
if the len is always 1, can we just make it a pointer to a location or something similar ?

Reply:
Hmm, not sure I understand the question.  Can you rephrase or elaborate?